tutorials/09-set-up-physics-engine-tpe.md

```diff
@@ -0,0 +1,246 @@
+- `<FEATURES>.cc` for the "FeatureList" implementation corresponding to each of
+the \ref ignition::physics::Feature "Features" member functions, using the
+physics engine API to realize the feature behavior. For a list of common
+pre-defined features in Ignition Physics, please refer to
+\ref physicsplugin "Understanding the Physics Plugin" tutorial.
+- `<FEATURES_TEST>.cc` for unit tests of the "FeatureList".
+
+### Implement a feature using TPE's API
+
+It is important to implement `EntityManagementFeatures` \ref ignition::physics::FeatureList "FeatureList"
+for any physics plugin, in order to manage simulation objects. Hence, we will
+show an example implementing \ref ignition::physics::ConstructEmptyWorldFeature "ConstructEmptyWorldFeature"
+in `EntityManagementFeatures` for TPE.
+
+Basically, \ref ignition::physics::ConstructEmptyWorldFeature "ConstructEmptyWorldFeature"
+has a subclass \ref ignition::physics::Feature::Engine "Engine" defining
+`ConstructEmptyWorld` member function. This feature will construct an empty
+simulation world from using the physics engine API. We will display the feature
+implementation here for convenience:
+- [EntityManagementFeatures.hh](https://github.com/ignitionrobotics/ign-physics/blob/main/tpe/plugin/src/EntityManagementFeatures.hh):
+
+```cpp
+#include <string>
+
+#include <ignition/physics/ConstructEmpty.hh>
+#include <ignition/physics/Shape.hh>
+#include <ignition/physics/GetEntities.hh>
+#include <ignition/physics/RemoveEntities.hh>
+#include <ignition/physics/Implements.hh>
+
+#include "Base.hh"
+
+namespace ignition {
+namespace physics {
+namespace tpeplugin {
+
+struct EntityManagementFeatureList : FeatureList<
+  GetEngineInfo,
+  GetWorldFromEngine,
+  GetModelFromWorld,
+  GetLinkFromModel,
+  GetShapeFromLink,
+  RemoveEntities,
+  ConstructEmptyWorldFeature,
+  ConstructEmptyModelFeature,
+  ConstructEmptyLinkFeature,
+  CollisionFilterMaskFeature
+> { };
+
+class EntityManagementFeatures :
+  public virtual Base,
+  public virtual Implements3d<EntityManagementFeatureList>
+{
+  ... // other features' member functions
+
+  // ----- Construct empty entities -----
+  public: Identity ConstructEmptyWorld(
+    const Identity &_engineID, const std::string &_name) override;
+
+  ... // other features' member functions
+};
+
+}
+}
+}
+```
+
+Together with other \ref ignition::physics::Feature "Features", the \ref ignition::physics::ConstructEmptyWorldFeature "ConstructEmptyWorldFeature"
+is included in `EntityManagementFeatureList` "FeatureList" to declare the related
+features for entity management purpose.
+
+The `EntityManagementFeatures` "FeatureList" here inherits from:
+- [Base](https://github.com/ignitionrobotics/ign-physics/blob/main/tpe/plugin/src/Base.hh)
+class for foundation metadata definitions of Models, Joints, Links, and Shapes objects
+of TPE to provide easy access to [tpelib](https://github.com/ignitionrobotics/ign-physics/tree/main/tpe/lib)
+structures in the TPE library.
```

Comment:
Just an additional note, `Base.hh` is not necessarily needed if there is a straightforward way to interface external physics engine class objects with ign-physics class objects. It's an optional file  and is up to the developer on how to implement it.